hmpps-kotlin-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/kotlin/auth/HmppsWebClientConfiguration.kt

```diff
@@ -45,20 +45,39 @@ private const val DEFAULT_HEALTH_TIMEOUT_SECONDS: Long = 2
 @ConditionalOnBean(ClientRegistrationRepository::class)
 @Configuration
 class HmppsWebClientConfiguration {
+
+  /**
+   * This method generates an instance of the [AuthorizedClientServiceOAuth2AuthorizedClientManager]
+   * class configured to cache all OAuth2 tokens under a single **principalName** using the
+   * [GlobalPrincipalOAuth2AuthorizedClientService].
+   *
+   * The purpose of this [OAuth2AuthorizedClientManager] is to avoid unnecessary token requests to HMPPS Auth,
+   * and it should be used for web clients where a username is being injected to the token request.
+   *
+   * @param clientRegistrationRepository
+   * @param oAuth2AuthorizedClientProvider
+   */
   @ConditionalOnMissingBean
   @Bean
   fun authorizedClientManager(
     clientRegistrationRepository: ClientRegistrationRepository,
-    oAuth2AuthorizedClientService: OAuth2AuthorizedClientService,
     oAuth2AuthorizedClientProvider: OAuth2AuthorizedClientProvider,
-  ): OAuth2AuthorizedClientManager = AuthorizedClientServiceOAuth2AuthorizedClientManager(
-    clientRegistrationRepository,
-    oAuth2AuthorizedClientService,
-  ).kotlinApply { setAuthorizedClientProvider(oAuth2AuthorizedClientProvider) }
+  ): OAuth2AuthorizedClientManager {
+    val globalPrincipalOAuth2AuthorizedClientService =
```

Comment:
Would it be better to allow this to be overridable? e.g. create this as a `@Bean` with `@ConditionalOnMissingBean`. Then it will be the default unless people choose otherwise - hard to predict future requirements.